middle_end/flambda2/to_cmm/to_cmm_expr.ml

```diff
@@ -93,9 +93,10 @@ let translate_apply0 env res apply =
       Lambda.Rc_normal
     | Nontail -> Lambda.Rc_nontail
   in
+  let param_arity = Apply.args_arity apply in
+  let return_arity = Apply.return_arity apply in
```

Comment:
Now that we have `param_arity` and `return_arity` here, we can compute the types and return types in all cases below properly.

Reply:
Should be fixed